gkr/src/executor.rs

```diff
@@ -190,43 +193,64 @@ pub async fn run_command<'a, Cfg: GKRConfig>(command: &ExpanderExecArgs, mut con
             witness_file,
             output_proof_file,
         } => {
-            let (mut circuit, mut window) = Circuit::<Cfg::FieldConfig>::prover_load_circuit::<Cfg>(
-                &circuit_file,
-                &config.mpi_config,
-            );
-            circuit.prover_load_witness_file(&witness_file, &config.mpi_config);
-            let (claimed_v, proof) = prove(&mut circuit, &config);
+            let (mut circuit, mut window) =
+                Circuit::<Cfg::FieldConfig>::prover_load_circuit::<Cfg>(&circuit_file, mpi_config);
+            let mpi_config = MPIConfig::prover_new();
+            let prover = Prover::<Cfg>::new(mpi_config.clone());
 
-            if config.mpi_config.is_root() {
+            circuit.prover_load_witness_file(&witness_file, &mpi_config);
+            let (claimed_v, proof) = prove::<Cfg>(&mut circuit, mpi_config.clone());
+
+            if prover.mpi_config.is_root() {
                 let bytes = dump_proof_and_claimed_v(&proof, &claimed_v)
                     .expect("Unable to serialize proof.");
                 fs::write(output_proof_file, bytes).expect("Unable to write proof to file.");
             }
             circuit.discard_control_of_shared_mem();
-            config.mpi_config.free_shared_mem(&mut window);
+            mpi_config.free_shared_mem(&mut window);
         }
         ExpanderExecSubCommand::Verify {
             circuit_file,
             witness_file,
             input_proof_file,
             mpi_size,
         } => {
-            assert!(
-                config.mpi_config.world_size() == 1,
-                "Do not run verifier with mpiexec."
-            );
-            config.mpi_config.world_size = mpi_size as i32;
+            let mpi_config = MPIConfig::verifier_new(mpi_size as i32);
+            let verifier = Verifier::<Cfg>::new(mpi_config);
+
+            // this assertion is not right: the MPI size = 2 so that the verifier knows the prover
+            // is running in 2 threads. The verifier itself is running in 1 thread.
+            //
+            // assert!(
+            //     verifier.mpi_config.world_size() == 1,
+            //     "Do not run verifier with mpiexec."
+            // );
+
```

Comment:
a bit of background and history - verifiers should not be run in MPI, yet the current verifier config still carry a MPI config instance, and verifier should be aware of the MPI world size in verififying, or it won't make it.  So I am fine with just deprecating and deleting the prev impl, but a bit of context clarification is still needed...